pkg/scheduler/framework/plugins/podtopologyspread/filtering_test.go

```diff
@@ -515,16 +514,17 @@ func TestPreFilterState(t *testing.T) {
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			ctx := context.Background()
-			informerFactory := informers.NewSharedInformerFactory(fake.NewSimpleClientset(tt.objs...), 0)
-			pl := PodTopologySpread{
-				sharedLister:       cache.NewSnapshot(tt.existingPods, tt.nodes),
-				defaultConstraints: tt.defaultConstraints,
+			snapshot := cache.NewSnapshot(tt.existingPods, tt.nodes)
+			n := func(plArgs runtime.Object, fh framework.Handle) (framework.Plugin, error) {
+				return New(plArgs, fh)
+			}
+			args := &config.PodTopologySpreadArgs{
+				DefaultConstraints: tt.defaultConstraints,
+				DefaultingType:     config.ListDefaulting,
 			}
-			pl.setListers(informerFactory)
-			informerFactory.Start(ctx.Done())
-			informerFactory.WaitForCacheSync(ctx.Done())
+			p := plugintesting.SetupPluginWithInformers(ctx, t, n, args, snapshot, tt.objs)
```

Comment:
you can just pass `New` to `SetupPluginWithInformers` in this case. Same for `SetupPlugin` below

Reply:
Done.